draft-ietf-quic-transport.md

```diff
@@ -345,6 +345,31 @@ the relative priority of streams.  When deciding which streams to dedicate
 resources to, the implementation SHOULD use the information provided by the
 application.
 
+## Required Operations on Streams
+
+There are certain operations which an application MUST be able to perform when
+interacting with QUIC streams.  This document does not specify an API, but
+any implementation of this version of QUIC MUST expose the ability to perform
+the operations described in this section on a QUIC stream.
+
+On the sending part of a stream:
+
+- Attempt to write data, understanding when stream flow control credit
+  ({{data-flow-control}}) has successfully been reserved to send the written
+  data or possibly discovering that the stream has been closed because the peer
+  sent a STOP_SENDING frame ({{frame-stop-sending}})
+- Cleanly terminate the stream, resulting in a STREAM frame ({{frame-stream}})
+  with the FIN bit set
+- Abruptly terminate the stream, resulting in a RESET_STREAM frame
+  ({{frame-reset-stream}}), even if the stream was cleanly terminated previously
+
+On the receiving part of a stream:
+
+- Attempt to read data, possibly discovering that the peer has terminated the
+  stream either cleanly or abruptly
+- Abort reading of the stream and request closure, resulting in a STOP_SENDING
+  frame ({{frame-stop-sending}})
+
 
```

Comment:
This could be read as if aborting a stream should always result in a STOP_SENDING frame, but I don't think that that is necessarily true. I understand that the ability to do so is important though.

Reply:
It's always true if the recipient hasn't already received through the end of the stream.  But yes, I'll try to expand on it.  It might be worth reframing some things in the transport doc to work in terms of acting on these requests from the application.